diff --git a/dist/build/index.js b/dist/build/index.js
index c26548ec48acc01070f20800cd3fee5793eba131..e3b8bf060678c1abec8cf6eacaff789b9c72baa8 100644
--- a/dist/build/index.js
+++ b/dist/build/index.js
@@ -2592,6 +2592,42 @@ async function build(dir, experimentalAnalyze = false, reactProductionProfiling
             if (proxyFilePath && bundler !== _bundler.Bundler.Turbopack) {
                 await _fs.promises.rename(_path.default.join(distDir, _constants1.SERVER_DIRECTORY, 'proxy.js'), _path.default.join(distDir, _constants1.SERVER_DIRECTORY, 'middleware.js'));
                 await _fs.promises.rename(_path.default.join(distDir, _constants1.SERVER_DIRECTORY, 'proxy.js.nft.json'), _path.default.join(distDir, _constants1.SERVER_DIRECTORY, 'middleware.js.nft.json'));
+                const middlewareNft = JSON.parse(
+                  await _fs.promises.readFile(
+                    _path.default.join(
+                      distDir,
+                      _constants1.SERVER_DIRECTORY,
+                      'middleware.js.nft.json'
+                    ),
+                    'utf8'
+                  )
+                )
+
+                // During build, tools like Sentry wrap user code in templates that import the compiled
+                // proxy.js file. This causes Next.js to trace 'proxy.js' into the NFT file. However,
+                // since we rename 'proxy.js' to 'middleware.js', deployment providers like Vercel
+                // will fail when checking if the 'proxy.js' file exists based on the NFT file.
+                let hasProxyJsInNft = false
+                middlewareNft.files = middlewareNft.files.map(
+                  (file) => {
+                    if (file === 'proxy.js') {
+                      hasProxyJsInNft = true
+                      return 'middleware.js'
+                    }
+                    return file
+                  }
+                )
+
+                if (hasProxyJsInNft) {
+                  await _fs.promises.writeFile(
+                    _path.default.join(
+                      distDir,
+                      _constants1.SERVER_DIRECTORY,
+                      'middleware.js.nft.json'
+                    ),
+                    JSON.stringify(middlewareNft)
+                  )
+                }
             }
             if (isCompileMode) {
                 _log.info(`Build ran with "compile" mode, to finalize the build run either "generate" or "generate-env" mode as well`);
